test/Transactions/Orleans.Transactions.Tests/Grains/MultiStateTransactionalGrains.cs

```diff
@@ -0,0 +1,144 @@
+﻿using System;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using Orleans.Transactions.Abstractions;
+
+namespace Orleans.Transactions.Tests
+{
+    [Serializable]
+    public class GrainData
+    {
+        public int Value { get; set; }
+    }
+
+    public class EightStateTransactionalGrain : MultiStateTransactionalGrainBaseClass
```

Comment:
Goal was to support TransactionTestConstants.MaxCoordinatedTransactions, which is currently 8 but may be redefined in the future.
This should be possible by injecting the Transactional State factory into the constructor rather than the specific state objects, and creating as many as is needed.